core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/runtimeModels.scala

```diff
@@ -311,7 +312,12 @@ object UserScriptPath {
   def stringToUserScriptPath(string: String): Either[Throwable, UserScriptPath] =
     parseGcsPath(string) match {
       case Right(value) => Right(Gcs(value))
-      case Left(_)      => Either.catchNonFatal(new URL(string)).map(url => Http(url))
+      case Left(_) =>
+        if (UrlValidator.getInstance().isValid(string)) {
```

Comment:
you can use `org.http4s.Uri` instead since the project already has it and we don't have to bring in additional dependency

```
@  org.http4s.Uri.fromString("http://127.0.0.1") 
res2: org.http4s.package.ParseResult[org.http4s.Uri] = Right(
  value = Uri(scheme = Some(value = Scheme(http)), authority = Some(value = Authority(userInfo = None, host = Ipv4Address(address = 127.0.0.1), port = None)), path = , query = , fragment = None)
)

@ org.http4s.Uri.fromString("userscript.sh | nslookup http://another-url.com") 
res3: org.http4s.package.ParseResult[org.http4s.Uri] = Left(value = ParseFailure(sanitized = "Invalid URI", details = "Error(13,NonEmptyList(EndOfString(13,47)))"))
```